tests/crypto/mbedtls_psa/testcase.yaml

```diff
@@ -0,0 +1,37 @@
+# The goal here is to showcase that Mbed TLS's PSA implementation can be
+# supported on any platform. A very minimal configuration is tested here: simply
+# enable support for PSA APIs in Mbed TLS and then test them with:
+# - psa_crypto_init() which is required before any PSA operation
+# - psa_generate_random() which is always available as long as PSA is initialized.
+#
+# Since it might take too long to really execute this test on all Zephyr's
+# supported platforms, we reduce to a very small selected group whose
+# characteristics are:
+# - no TF-M enabled devices because we assume that the TF-M implementation
+#   of PSA is working fine on the platforms that support TF-M.
+# - platform should be testable by the CI.
+# - enable CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG in order to reduce as much
+#   as possible usage of legacy modules in Mbed TLS.
+# - pick 1 platform which supports entropy driver and 1 which does not. The
+#   latter case will allow to test
+#   CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG_ALLOW_NON_CSPRNG.
+common:
+  tags:
+    - mbedtls
+    - psa
+tests:
+  crypto.mbedtls_psa.with_entropy_driver:
+    # Pick a platform which has an entropy driver and enable it.
+    platform_allow:
```

Comment:
I think that instead of using `platform_allow` you could use `integration_platforms`. Especially since the behavior of `platform_allow` may be surprising in certain cases (https://github.com/zephyrproject-rtos/zephyr/pull/75962). See https://docs.zephyrproject.org/latest/develop/test/twister.html for reference.

Reply:
Thanks for the tip! I didn't know the difference between `platform_allow` and `integration_platform` before.